src/zero_copy/combinator.rs

```diff
@@ -959,6 +959,96 @@ where
     go_extra!();
 }
 
+#[derive(Copy, Clone)]
+pub struct Not<A> {
+    pub(crate) parser: A,
+}
+
+impl<'a, I, E, S, A> Parser<'a, I, E, S> for Not<A>
+where
+    I: Input + ?Sized,
+    E: Error<I>,
+    S: 'a,
+    A: Parser<'a, I, E, S>,
+{
+    type Output = I::Token;
+
+    fn go<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E, S>) -> PResult<M, Self::Output, E> {
+        let before = inp.save();
+
+        let result = self.parser.go::<Check>(inp);
+        inp.rewind(before);
+
+        let (at, tok) = inp.next();
+        match result {
+            Ok(_) => Err(Located::at(
+                at,
+                E::expected_found(None, tok, inp.span_since(before)),
+            )),
+            Err(_) => {
+                if let Some(tok) = tok {
+                    Ok(M::bind(|| tok))
+                } else {
+                    Err(Located::at(
+                        at,
+                        E::expected_found(None, None, inp.span_since(before)),
+                    ))
+                }
```

Comment:
As discussed in #208, I think this case is probably more problematic than it's worth and `not` should instead parse no tokens at all on success.